app/http.php

```diff
@@ -0,0 +1,313 @@
+
+/**
+ * @param array<string, mixed> $state
+ */
+function markOnline(string $executorHostname, array $state, bool $forceShowError = false): void
+{
+    global $executorStates;
+
+    $oldState = $executorStates->exists($executorHostname) ? $executorStates->get($executorHostname) : [];
+
+    $state['status'] = 'online';
+    $state['hostname'] = $executorHostname;
+    $state['state'] = \json_encode($state);
+    $executorStates->set($executorHostname, $state);
+
+    if (!$oldState || ($oldState['status'] ?? '') === 'offline' || $forceShowError) {
+        Console::success('Executor "' . $executorHostname . '" went online.');
+    }
+}
+
+function fetchExecutorsState(bool $forceShowError = false): void
+{
+    $executors = \explode(',', App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTORS', ''));
+
+    foreach ($executors as $executor) {
+        go(function () use ($executor, $forceShowError) {
+            try {
+                $endpoint = 'http://' . $executor . '/v1/health';
+
+                $ch = \curl_init();
+
+                \curl_setopt($ch, CURLOPT_URL, $endpoint);
+                \curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+                \curl_setopt($ch, CURLOPT_TIMEOUT, 10);
+                \curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
+                \curl_setopt($ch, CURLOPT_HTTPHEADER, [
+                    'x-open-runtimes-executor-secret: ' . App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTOR_SECRET', '')
+                ]);
+
+                $executorResponse = \curl_exec($ch);
+                $statusCode = \curl_getinfo($ch, CURLINFO_HTTP_CODE);
+                $error = \curl_error($ch);
+
+                \curl_close($ch);
+
+                if ($statusCode == 200 && !\is_bool($executorResponse)) {
+                    $body = (array) \json_decode($executorResponse, true);
+
+                    if ($body['status'] === 'pass') {
+                        markOnline($executor, $body, $forceShowError);
+                    } else {
+                        $message = 'Response does not include "pass" status.';
+                        markOffline($executor, $message, $forceShowError);
+                    }
+                } else {
+                    $message = 'Code: ' . $statusCode . ' with response "' . $executorResponse .  '" and error error: ' . $error;
+                    markOffline($executor, $message, $forceShowError);
+                }
+            } catch (\Exception $err) {
+                throw $err;
+            }
+        });
+    }
+}
+
+/**
+ * Create logger instance
+ */
+$providerName = App::getEnv('OPEN_RUNTIMES_PROXY_LOGGING_PROVIDER', '');
+$providerConfig = App::getEnv('OPEN_RUNTIMES_PROXY_LOGGING_CONFIG', '');
+$logger = null;
+
+if (!empty($providerName) && !empty($providerConfig) && Logger::hasProvider($providerName)) {
+    $classname = '\\Utopia\\Logger\\Adapter\\' . \ucfirst($providerName);
+
+    /**
+     * @var Adapter $adapter
+     */
+    $adapter = new $classname($providerConfig);
```

Comment:
Might be better to use a map or a switch case to avoid unexpected env vars.

Reply:
I think we did this on purpose. So when a new adapter is introduced in utopia/logger, it is automatically picked up. Doing switch/case here would add extra maintenance we would need to do each time we upgrade the logger.

Let me know if you still want to use the switch/case here.

Reply:
I think we might want more control at this stage for which adapters are allowed until we feel more comfortable that these mechanism are flexible enough.

Even then, we might want to ask the library which adapters are actually supported so we can throw proper exception, this kind of method should probably be consistent across Utopia libraries. Not something I would worry about right now.